backend/src/Service/Collection/CollectionService.php

```diff
@@ -14,12 +18,203 @@ public function __construct(
     {
     }
 
+    public function findByUuid(string $uuid): ?Collection
+    {
+        try {
+            $uuidObject = Uuid::fromString($uuid);
+            return $this->em->getRepository(Collection::class)->findOneBy(['uuid' => $uuidObject]);
+        } catch (\InvalidArgumentException $e) {
+            return null;
+        }
+    }
+
     /**
      * @return Collection[]
      */
-    public function getUserCollections(): array
+    public function getUserCollections(int $hyvorUserId): array
+    {
+        $ownedCollections = $this->em->getRepository(Collection::class)->findBy([
```

Comment:
Currently, this is a special case.

Make it so that default collections and owned collections also have collection_user records for access. So, we only need the below one.